protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java

```diff
@@ -36,6 +38,77 @@
 public class HintRetrieveRequestTest {
 
     @Test
+    public void testFromAuthMethods() {
+        HintRetrieveRequest request = HintRetrieveRequest.fromAuthMethods(
+                AuthenticationMethods.EMAIL,
+                AuthenticationMethods.FACEBOOK);
+
+        assertThat(request.getAuthenticationMethods().size()).isEqualTo(2);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.EMAIL);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.FACEBOOK);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testFromAuthMethods_noneProvided() {
+        HintRetrieveRequest.fromAuthMethods();
+    }
+
+    @Test
+    public void testFromAuthMethods_withSet() {
+        HintRetrieveRequest request = HintRetrieveRequest.fromAuthMethods(
+                ImmutableSet.of(AuthenticationMethods.EMAIL, AuthenticationMethods.FACEBOOK));
+
+        assertThat(request.getAuthenticationMethods().size()).isEqualTo(2);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.EMAIL);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.FACEBOOK);
+    }
+
+    @Test(expected=IllegalArgumentException.class)
+    public void testFromAuthMethods_emptySet() {
+        HintRetrieveRequest request = HintRetrieveRequest.fromAuthMethods(
+                Collections.emptySet());
+
+        assertThat(request.getAuthenticationMethods().size()).isEqualTo(2);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.EMAIL);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.FACEBOOK);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testBuilder_emptyAuthMethodsSet() {
+        new HintRetrieveRequest.Builder(Collections.emptySet());
+    }
+
+    @Test
+    public void testBuilder_setTokenProviders() {
+        HintRetrieveRequest request = new HintRetrieveRequest.Builder(AuthenticationMethods.EMAIL)
+                .setTokenProviders(ValidTokenProviderMap.make())
+                .build();
+
+        ValidTokenProviderMap.assertEquals(request.getTokenProviders());
+    }
+
+    @Test
+    public void testBuilder_setTokenProviders_nullMap() {
+        HintRetrieveRequest request = new HintRetrieveRequest.Builder(AuthenticationMethods.EMAIL)
+                .setTokenProviders(null)
+                .build();
+
+        assertThat(request.getTokenProviders()).isEmpty();
+    }
+
+    @Test
+    public void testBuilder_addTokenProvider_noInfo() {
+        String tokenProvider = "https://my.token.provider";
+        HintRetrieveRequest request = new HintRetrieveRequest.Builder(AuthenticationMethods.EMAIL)
+                .addTokenProvider(tokenProvider)
+                .build();
+
+        assertThat(request.getTokenProviders().containsKey(tokenProvider));
```

Comment:
Missing an assertion here.

Reply:
misplaced bracket, containsKey is an both a method on maps and on the map assertions :/